src/jit/instrsxarch.h

```diff
@@ -192,7 +192,6 @@ INST3( xorps,       "xorps"       , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, PCK
 
 INST3( cvttsd2si,   "cvttsd2si"   , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, SSEDBL(0x2C)) // cvt with trunc scalar double to signed DWORDs
 
-#ifndef LEGACY_BACKEND
```

Comment:
I think we should leave the instructions available to the `LEGACY_BACKEND` the same. That is, don't remove this (and other, below) `#ifdefs`.

We don't want the x86 legacy backend to start generating new instructions.

Reply:
I realize this might require compensating code somewhere else, such as in the emitter.

Reply:
e.g., `IsSSEOrAVXInstruction` and similar should probably follow the model of `IsSSE4Instruction` and return false for `LEGACY_BACKEND`

Reply:
I tried to retain `#ifndef LEGACY_BACKEND` here, but that may require replacing most of `#ifdef FEATURE_AVX_SUPPORT` by `#ifndef LEGACY_BACKEND`.

Reply:
> I tried to retain #ifndef LEGACY_BACKEND here, but that may require replacing most of #ifdef FEATURE_AVX_SUPPORT by #ifndef LEGACY_BACKEND.

I agree with @BruceForstall that it is risky to add instructions to `LEGACY_BACKEND`; I would think that most of the impact would be in the code that has lists of instructions, such as the ones @BruceForstall mentions above.

Reply:
Additionally, if we keep these instructions under `#ifndef LEGACY_BACKEND`, we would not need to change the `code_t` definition https://github.com/dotnet/coreclr/issues/14065#issuecomment-333692347

Reply:
> Additionally, if we keep these instructions under #ifndef LEGACY_BACKEND, we would not need to change the code_t definition

That would be a good thing.